dc-commons-xml/src/main/java/de/digitalcollections/commons/xml/xpath/XPathMapper.java

```diff
@@ -127,6 +135,26 @@ public XPathMapper(Class<T> targetType, String[] rootPaths, String defaultRootNa
     }
   }
 
+  /**
+   * Create a new instance of the target type from the given XML document located at pathToXmlFile.
+   */
+  public T readDocument(Path pathToXmlFile)
+      throws XPathMappingException, IOException, ParserConfigurationException, SAXException {
+    try (InputStream inputStream = Files.newInputStream(pathToXmlFile)) {
+      return readDocument(inputStream);
+    }
+  }
+
+  /** Create a new instance of the target type from the given XML document by its InputStream. */
+  public T readDocument(InputStream inputStream)
+      throws XPathMappingException, ParserConfigurationException, IOException, SAXException {
+    DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
```

Comment:
Das kann statisch sein, oder?

Reply:
nein: public static T readDocument(...): compile-Fehler: "non-static type variable T cannot be referenced from a static context"

Reply:
Nein, ich mein nicht die Methode sondern die `DocumentBuilderFactory` instanz, die muss doch nicht jedes mal neu aufgezogen werden sondern kann einmaal statisch initialisiert werden und gut ist.

Reply:
ah, sorry, bezieht sich natürlich immer auf die exakt darüberstehende Zeile....

Reply:
dann doch gleich den DocumentBuilder static, oder?

Reply:
Siehe anderer Kommentar, `DocumentBuilder` ist nicht thread-safe scheinbar. https://github.com/dbmdz/digitalcollections-commons/pull/253#discussion_r404597626

Reply:
ok, erledigt